pallets/eth-bridge/src/lib.rs

```diff
@@ -824,28 +830,30 @@ pub mod pallet {
     }
 
     // TODO: re-add the `Accepted and Rejected events
-    fn process_ethereum_event<T: Config>(event: &EthEvent) -> Result<(), DispatchError> {
-        // TODO before processing ensure that the event has not already been processed
-        // Do the check that is not processed via ProcessedEventsChecker
-
-        ensure!(false == T::ProcessedEventsChecker::processed_event_exists(&event.event_id.clone()), Error::<T>::EventAlreadyProcessed);
-        let mut event_accepted = false;
-
-        match T::BridgeInterfaceNotification::on_incoming_event_processed(&event) {
-            Ok(_) => { event_accepted = true },
-            Err(err) => {
-                log::error!("💔 Processing ethereum event failed: {:?}", err);
-            },
-        };
+    fn process_ethereum_event<T: Config>(event: &EthEvent) {
+        if T::ProcessedEventsChecker::processed_event_exists(&event.event_id.clone()) {
+            log::error!("💔 Event already processed, duplicate event");
 
-        <Pallet<T>>::deposit_event(Event::<T>::EventProcessed {
-            accepted: event_accepted,
-            eth_event_id: event.event_id.clone(),
-        });
-        // Add record of succesful processing via ProcessedEventsChecker
-        T::ProcessedEventsChecker::add_processed_event(&event.event_id.clone(), event_accepted);
+            <Pallet<T>>::deposit_event(Event::<T>::DuplicateEventSubmission {
```

Comment:
I see why you are emitting an event here, I am assuming its because we don't want to error here which is correct. But in this particular case, the event we are processing has already been processed so Its "safe" to return an error and undo any storage changes related to processing this event.